src/ui/wxWidgets/GridTable.cpp

```diff
@@ -162,6 +162,24 @@ void GridTable::SetView(wxGrid* newGrid)
       else
         newGrid->HideCol(idx);
 #endif
+      
+      if(PWSprefs::GetInstance()->GetPref(PWSprefs::OptimizedCellSize)) {
+        //Determin best fit column width
+        wxSize borderSize = m_pwsgrid->GetWindowBorderSize();
+        wxScreenDC dc;
+        wxCoord width, height, border = (borderSize.GetWidth() * 2) + 2;
+        dc.SetFont(m_pwsgrid->GetFont());
+      
+        for(unsigned int row = 0, noRows = GetNumberRows(); row < noRows; ++row) {
+          if(!IsEmptyCell(row, idx)) {
+            dc.GetTextExtent(GetValue(row, idx), &width, &height);
+            width += border;
+            if(width > PWSGridCellData[idx].width) {
+              PWSGridCellData[idx].width = width;
+            }
+          }
+        }
+      }
```

Comment:
Would it be an option to use [wxGrid::AutoSizeColumns](https://docs.wxwidgets.org/3.0/classwx_grid.html#a412837fe2932fcde8473e6fbf64f8ad5) and let the framework do the calculation of the best column width?

Reply:
I did not managed to adapt the code using wxGrid::AutoSizeColumns to have the expected functionality.

Reply:
Hm, an event handler for double-click events is set up in the GridCtrl [constructor](https://github.com/pwsafe/pwsafe/blob/0e789f083e1f3dfff8254af819642d1dd4ed03f2/src/ui/wxWidgets/GridCtrl.cpp#L66), from which [wxGrid::AutoSizeColumn](https://github.com/pwsafe/pwsafe/blob/0e789f083e1f3dfff8254af819642d1dd4ed03f2/src/ui/wxWidgets/GridCtrl.cpp#L81) is already successfully used when a double-click event is executed on one of the column separators.

Reply:
Had to call function AusSizeColumns outside of the loop.